src/AutoRest.CSharp/Common/Generation/Writers/ClientWriter.cs

```diff
@@ -16,7 +16,7 @@ internal class ClientWriter
         protected const string PipelineVariable = "pipeline";
         protected const string PipelineProperty = "Pipeline";
         protected const string PipelineField = "_" + PipelineVariable;
-        protected const string DiagnosticOptionsProperty = "DiagnosticOptions";
+        protected const string DiagnosticsProperty = "Diagnostics";
```

Comment:
I don't see `Rename Provider to ResourceProvider` changes in this PR.  I only see DiagnosticOptions -> Diagnostics and BaseUri -> Endpoint

Reply:
yeah, we don't have anything that uses it, but basically the change is to have autorest reference the new version of Azure.ResourceManager and through that it picks up the new definition of Provider.